app/workers/vendor_api_request_worker.rb

```diff
@@ -0,0 +1,37 @@
+class VendorAPIRequestWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: 3, queue: :low_priority
+
+  def perform(request_data, response_body, status_code, created_at)
+    request_headers = request_data['headers']
+    provider_id = provider_id_from_auth_token(request_headers.delete('HTTP_AUTHORIZATION'))
+
+    VendorAPIRequest.create!(
+      request_path: request_data['path'],
+      request_headers: request_headers,
+      request_body: request_data['params'],
+      response_body: response_hash(response_body, status_code),
+      status_code: status_code,
+      provider_id: provider_id,
+      created_at: created_at,
+    )
+  end
+
+private
+
+  def provider_id_from_auth_token(auth_header)
+    return if auth_header.blank?
+
+    token = auth_header[7..-1]
+    hashed_token = Devise.token_generator.digest(VendorAPIToken, :hashed_token, token)
+    VendorAPIToken.find_by(hashed_token: hashed_token)&.provider_id
```

Comment:
You could use `VendorAPIToken.find_by_unhashed_token` here, which does the hashing of the token.

Reply:
:+1: https://github.com/DFE-Digital/apply-for-teacher-training/pull/3009/files#diff-ea82dc5cde409c49f45d4efc687841c9R29-R30